### ISSUE TITLE
Diffusion model cannot accept conditioned / start-end inputs (conditional & route generation crash; geometry ignored)

### ISSUE THREAD

**Issue by ionfwsrijan** [KanishJebaMathewM/Truxify#13876]
﻿## Description

`backend/ml/diffusion/model.py` concatenates `condition`/`start_end` onto the input before the first linear layer, whose `in_features` is fixed at `input_dim`:

```python
# model.py:108
self.input_proj = nn.Linear(input_dim, hidden_dim)
# model.py:199 (denoise)
if condition is not None:
    x_t = torch.cat([x_t, condition], dim=-1)      # last dim becomes input_dim + cond_dim
# model.py:249-252 (generate_route)
route_input = torch.cat([start_point, end_point], dim=-1)   # last dim = 2*input_dim
route = self.generate(route_input.shape, condition)
```

`self.model` (built with `nn.Linear(input_dim, hidden_dim)`) then receives a tensor whose last dim is **not** `input_dim`.

## Expected Behavior

Conditional inference and route generation embed the condition/start-end into the model's hidden dimension (e.g. a separate `cond_proj` added to `x`), and `generate_route` actually constrains generation to the requested endpoints.

## Actual Behavior

1. Every `/diffusion/conditional` or `/diffusion/generate-route` call crashes with a `Linear` size mismatch.
2. Even if it ran, `generate_route` **never uses** `start_point`/`end_point` to constrain the output — it only borrows their `.shape`, so the "route" is an unconstrained sample unrelated to the requested endpoints. The feature is both broken and semantically meaningless.

## Proposed Fix

```python
self.cond_proj = nn.Linear(cond_dim, hidden_dim) if cond_dim else None
# in denoise: do NOT concat; add a projected condition
if condition is not None:
    c = self.cond_proj(condition)
    x = x + c.unsqueeze(1)
```
For `generate_route`, inject `start_point`/`end_point` as boundary conditioning rather than only reusing their shape; if they aren't used, remove the misleading concat. Add tests for conditional and route-generation paths.
